### ISSUE TITLE
Input events from Wacom pen aren't working

### ISSUE THREAD

**Issue by CharlieHess** [electron/electron#7815]
* Electron version: 1.3.7+
* Operating system: macOS

We've gotten a few reports of Wacom pens not working in Electron apps:

>I had trouble to using my wacom tablet on slack apps for clicking, scrolling etc. It doesn't register the click etc. [...] all my other electron wrapped apps has the same problem as this. Franz, Paws for Trello, etc

Etc.

>I use a Wacom pen and tablet as an input device. As of the most recent two updates of Slack, the tablet no longer functions properly in Slack. [...] I am running Mac OS X El Capitan (10.11.6), Slack 2.3.1 and Wacom driver 6.3.18-4 on a MacBook Pro Retina, 15-inch, Mid 2015 and Wacom Intuos5 Touch tablet.

We don't have a Wacom device here to test yet but we'll order one.

**Comment by kevinsawicki** [comment]
> We don't have a Wacom device here to test yet but we'll order one.

Any updates on this @CharlieHess? Have you been able to test it out on a Wacom?

**Comment by chrleon** [comment]
Any updates on this @CharlieHess?

**Comment by asacui** [comment]
I have the same problem. I have Intuos Pro Model PTH-851. I am running the latest Wacom driver. Slack does not register all the pen clicks. There is some limited Wacom Slack interaction. The pen does not allow me to select type. There is no response to clicking on videos or images (cannot hit play, send image to full screen). Cannot switch between threads or direct messages. Scroll bar doesn't work. I believe the problem was not this bad several months ago. My Wacom usually works fine outside of Slack. On rare occasion it does not switch preferences from one application to the other (occasionally does not switch from Blender pen preferences to Finder preferences).

**Comment by kevinsawicki** [comment]
@asacui do you experience the same problems when using Slack in Chrome?

**Comment by asacui** [comment]
Slack on Chrome does not have the problem. Thank you.

**Comment by asacui** [comment]
I use Mac OS Sierra. 10.12.3.

**Comment by asacui** [comment]
Very interesting. I went back to the Slack app and the problem is fine. It might be that the problem doesn't happen every time. I guess I would have taken action earlier if the problem didn't sometimes not appear.

During the times when the interactivity is broken, it is reliably broken.

**Comment by kevinsawicki** [comment]
@asacui are you able to run `npm install electron@beta`, launch via the `electron` command then try out the default app? Does scrolling, clicking, etc. function correctly in it?

**Comment by asacui** [comment]
Slack only has interactivity difficulties with Wacom. Other input devices work fine.

I am new to github. Not exactly sure what the site is. Run npm on Terminal?

**Comment by kevinsawicki** [comment]
@asacui are you on Mac? If so, it is probably easier to just download https://github.com/electron/electron/releases/download/v1.6.0/electron-v1.6.0-darwin-x64.zip directly and launch it and see if the default app works as expected with the wacom input.

**Comment by asacui** [comment]
I'm on a work Mac. My company has tight rules about installing apps. Might take me a while to get permission to run Electron. 

I apologize. I'm not quite sure what the function of GitHub is. I thought perhaps it was a platform to let Slack know about the issue.

**Comment by asacui** [comment]
@kevinsawicki: I have Electron ready to go. I am a newbie at this.
![screen shot 2017-02-15 at 11 20 48 am](https://cloud.githubusercontent.com/assets/25774768/22983697/e633d410-f370-11e6-905a-57efe1c1b560.png)

**Comment by CaliLuke** [comment]
I can confirm the problem exists and it's pretty serious, slack is nearly unusable.

**Comment by BenjiBauer** [comment]
Hey, I got the same problem. I'm building an app with electron and I am using the myScript API for handwriting recognition. There are no problems with Myscript and my wacom Intous on other browsers or chrome itself. There is only a lag when i am using it with electron. Click and writing is possible but you have to rest the pen for maybe 100-500 Milliseconds on the same spot before you start writing. 
I am pretty new in electron..

Well I found out, that there are no problems with the pen input if you set experimentalFeatures true:
`mainWindow = new BrowserWindow({width: 800, height: 600, 'webPreferences': {'experimentalFeatures': true}});`

But the I have a way bigger problem. All my onmousedown, onmouseup etc. events are not working anymore! :(

**Comment by everyplace** [comment]
Citing the as still a problem. Looks like this article on how Wacom works with Chrome is worth investigating: 

https://discuss.atom.io/t/how-to-access-chrome-flags-in-electron/16470

My wacom pen does not work in Slack (but the tablet with finger input works just fine).

**Comment by jaap** [comment]
I'm experiencing similar issues. It seems to happen when an application is open and the tablet is connected while the app is running. Restarting the app brings back the pen-click functionality on my Mac.

**Comment by thorstenkonrad** [comment]
same issue that I have. 
Since half a year I always have to restart slack 20 per day. restart works than for 2-3 clicks. After that slack is not usable any longer for me. can't click. Application does not react on clicks. Tried already a lot of different wacom-driver. Also Tried different slack versions (beta, app-store, download-etc.). Same issue.

**Comment by sinspiral** [comment]
I'm suffering this too... but for me it is not only on Electron based apps, but in Chromium based apps in general. This happens randomly... sometimes just going to another app for a while and going back to Chromium solves it by itself. Even sometimes hitting the pen a couple of times makes it work, but this maybe just coincidence and biased by frustration ;-)

Some JS debugging got me to this point (see screenshot attached). I made this codepen in order to debug it: https://codepen.io/sinspiral/pen/GvEVPp?editors=0010

Looks like "mousedown" events are not getting triggered... but surprisingly "mouseup" did work (how can that even be possible?!?). So since those mousedown don't appear any "click" events don't happen.

Hope it this helps!!
![mouseup_only](https://user-images.githubusercontent.com/259592/29216186-ca4998fc-7e7b-11e7-987f-4aa299e02cf2.png)

**Comment by afkatja** [comment]
Like @sinspiral I'm having these issues in Chromium apps, but specifically, in Chrome dev tools. The "normal" clicks in Chrome are registered, but in dev tools, after 2-3 clicks, nothing happens.

**Comment by neostar** [comment]
Issue is probably connected to this Chromium bug report:
https://bugs.chromium.org/p/chromium/issues/detail?id=757069

**Comment by sofianguy** [comment]
@CharlieHess are you still running into this problem in later versions of electron?

**Comment by croesike** [comment]
I still run into this Problem with all kinds of Electron Apps.

**Comment by sofianguy** [comment]
@CharlieHess I'm going to close this issue because we are no longer fixing bugs older than 1.7.x 
Feel free to reopen if you find it is still a problem in later versions of Electron

**Comment by thorstenkonrad** [comment]
ok for me. never happened again.

**Comment by edipangi** [comment]
Retested the bug with the default app on version v27.0.4 on Fedora Linux 39. The bug is still present @sofianguy.

Steps to reproduce: 

1. Use a wacom graphics tablet for input.
2. Download default electron app.
3. Try to click any buttons using pen.
4. Observe that it is not working.

**Comment by chpio** [comment]
It's the same for me on a touchscreen-device when using the pen on linux+wayland.
